### PR TITLE
New package: clapack

### PR DESCRIPTION
**Pull request by fanne-stat**
New package CLAPACK

### CODE REVIEW

File: var/spack/repos/builtin/packages/clapack/package.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    obtain C code, and then modified to improve readability.  CLAPACK's goal
+    is to provide LAPACK for someone who does not have access to a Fortran
+    compiler."""
+
+    homepage = "http://www.netlib.org/clapack/"
+    url      = "http://www.netlib.org/clapack/clapack.tgz"
+
+    version('3.2.1', '040da31f3a7d4fbc9ac376c748d18d1f')
+
+    variant('external-blas', default=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/clapack/package.py]: Needs a description.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/clapack/package.py]: It's up to you. I agree that it probably has an ancient BLAS/LAPACK standard. But it's not like someone will accidentally use it, it isn't the default provider.

**Comment by adamjstewart** [comment]
It looks like a binary file got sucked into this PR.

File: var/spack/repos/builtin/packages/clapack/package.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    homepage = "http://www.netlib.org/clapack/"
+    url      = "http://www.netlib.org/clapack/clapack.tgz"
+
+    version('3.2.1', '040da31f3a7d4fbc9ac376c748d18d1f')
+
+    variant('external-blas', default=True,
+            description='Build with external BLAS (ATLAS here).')
+
+    depends_on('atlas', when='+external-blas')
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/clapack/package.py]: Can this be made into a virtual dependency?

File: var/spack/repos/builtin/packages/clapack/package.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    is to provide LAPACK for someone who does not have access to a Fortran
+    compiler."""
+
+    homepage = "http://www.netlib.org/clapack/"
+    url      = "http://www.netlib.org/clapack/clapack.tgz"
+
+    version('3.2.1', '040da31f3a7d4fbc9ac376c748d18d1f')
+
+    variant('external-blas', default=True,
+            description='Build with external BLAS (ATLAS here).')
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/clapack/package.py]: Do Spack provide any non-Fortran BLAS implementations?  If not, then making this default to True would defeat the stated purpose of CLAPACK.

File: var/spack/repos/builtin/packages/clapack/package.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                            'BLASLIB = ../../libcblaswr.a -lcblas -latlas')
+            makefile.filter(r'^lib.*',
+                            'lib: variants lapacklib tmglib')
+
+    def build(self, spec, prefix):
+        make('f2clib')
+        if '~external-blas' in spec:
+            make('blaslib')
+        elif '+external-blas' in spec:
+            make('cblaswrap')
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/clapack/package.py]: Can we do this with a simple if-then-else conditional?
```
make('cblaswrap' if '+external-blas' in spec else 'blaslib')
```

File: var/spack/repos/builtin/packages/clapack/package.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    version('3.2.1', '040da31f3a7d4fbc9ac376c748d18d1f')
+
+    variant('external-blas', default=True,
+            description='Build with external BLAS (ATLAS here).')
+
+    depends_on('atlas', when='+external-blas')
+
+    def edit(self, spec, prefix):
+        install('make.inc.example', 'make.inc')
+        make_inc = FileFilter('make.inc')
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/clapack/package.py]: Move this inside the if statement below.

File: var/spack/repos/builtin/packages/clapack/package.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    def build(self, spec, prefix):
+        make('f2clib')
+        if '~external-blas' in spec:
+            make('blaslib')
+        elif '+external-blas' in spec:
+            make('cblaswrap')
+        make('lib')
+
+    def install(self, spec, prefix):
+        copy_tree('.', prefix)
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/clapack/package.py]: @adamjstewart Should `install_tree()` or `copy_tree()` be used here?

File: var/spack/repos/builtin/packages/clapack/package.py (hunk 2 above)
Comment (fanne-stat) [reply var/spack/repos/builtin/packages/clapack/package.py]: CLAPACK in no longer maintained so I don't think we should make the built-in blas to be a virtual dependency. And this blas is not optimized for machine.

File: var/spack/repos/builtin/packages/clapack/package.py (hunk 3 above)
Comment (fanne-stat) [reply var/spack/repos/builtin/packages/clapack/package.py]: I did this according to the installation documentation of CLAPACK. They used atlas as external blas.

**Comment by fanne-stat** [comment]
@adamjstewart @citibeth @davydden Can clapack be merged now?

**Pull request merged**